remix/app/routes/_app.newsletter.update.$id.tsx

```diff
@@ -42,11 +44,31 @@ export async function action({ request, params }: ActionArgs) {
   });
 }
 
+export const loader = async ({ request, params }: ActionArgs) => {
+  return appHandler(request, async (user) => {
+    const newsletter = await NewsletterRepository.findFirst({
+      where: { id: params.id, userId: user.id },
+    });
+
+    return json({ ...newsletter });
+  });
+};
+
 export default function Index() {
   const result = useActionData();
+  const newsletter = useLoaderData();
   const navigate = useNavigate();
 
   useEffect(() => {
+    if (Object.keys(newsletter).length === 0) {
```

Comment:
Why do we need to check on the object keys? Is not the following sufficient?

```javascript
if (!newsletter) {

}
```

Reply:
Umm, this was my bad I was over engineer this.

Fixed here: [b357dd3](https://github.com/nimblehq/js-framework-benchmark/pull/130/commits/b357dd39fc2364fe1acf62523fa9f7463e88caed)